VERSIONING.md

```diff
@@ -0,0 +1,145 @@
+follows:
+
+- **UXP Version**: the entire semantic version for a UXP release (e.g.
+  `v1.2.2-up.3`).
+- **Crossplane Version**: the version of Crossplane embedded in the UXP release
+  (e.g. `v1.2.2`).
+- **Components Marker**: the version of the components surrounding upstream (or
+  temporarily forked) Crossplane (e.g. `-up.3`).
+
+## Guarantees
+
+![versioning-diagram](docs/media/versioning.png)
+
+The UXP versioning strategy can be expressed through a series of guarantees,
+each of which are elaborated in the subsequent sections. The diagram above
+illustrates examples of each guarantee in practice.
+
+1. The components marker on UXP main should always end in `-up.1` (i.e. no
+   components release candidates).
+2. The Crossplane version on UXP main must always be a release candidate version
+   for the first release of a major/minor pair (i.e. `vX.Y.0-rc.Z`).
+3. The Crossplane version on any `release-*` branch should _always_ be a
+   Crossplane release version (e.g. `v1.2.2`) (i.e. no release candidates).
+4. Not every Crossplane patch for a given `<major>`/`<minor>` version must be
+   included in a UXP release, even if the `<major>`/`<minor>` version has been
+   selected for long-term service (LTS). This would typically occur if
+   Crossplane made multiple patch releases in quick succession and a UXP version
+   had not been released for each in the intermediate (e.g. if `v1.2.4` is
+   released in upstream Crossplane and has not been included in a UXP release
+   before `v1.2.5` is released upstream, UXP may elect to forego cutting a
+   release with `v1.2.4`).
+5. At least one release candidate (RC) version should exist for any version that
+   is released. This is typically accomplished by cutting the RC version for the
+   _next_ release immediately after release of the preceding version.
+6. As a consequence of the preceding rule, a UXP version on a `release-*` branch
+   may end in `-rc.X` (e.g. `v1.2.2-up.4-rc.0`).
+7. A release _always_ results in the subsequent tag on the same branch to have
+   an incremented components version (e.g. the tag after `v1.2.2-up.2` must
+   include `-up.3`).
+8. A release for an embedded Crossplane version that has not yet been included
+   in a UXP release must have a components marker of `-up.1`.
+9. The components marker does not indicate the same components across Crossplane
+   versions or release branches. It is specific to the Crossplane version it is
+   appended to (e.g. `v1.2.2-up.1` does not indicate the same version of `xgql`
+   / `agent` / `bootstrapper` / etc. as `v1.3.0-up.1`).
+
+### Crossplane Version on Main
+
+The Crossplane version on the UXP repository `main` branch should always match
+that of the latest Crossplane tag on `master`. This ensures that we are always
+moving towards the next release branch cut that both Crossplane and UXP will
+make. By definition, this also means that the Crossplane version on UXP `main`
+will always be an RC version (e.g.`v1.4.0-rc.0-up.1`)
```

Comment:
Not a priority, but this seems like something we could automate pretty easily (e.g. with a cronjob style GitHub Action on the UXP repo).

Reply:
@negz totally agree :) it was in the original doc (as suggested by @muvaf), but didn't include here since this will be a long-lived doc. I'll make sure to open some issues after merge 👍🏻